examples/Custom_Hooks/callable.py

```diff
@@ -0,0 +1,92 @@
+## E2E Test for using callables
+
+# This example shows how to use the register_hook function to request internal data from different objects such as campaign, recommenders and objective.
+
+# To request internal data from the recommendation process this function can be used by giving it two arguments:
+# - The class method that contains the required data
+# - The callable function that will process the data
+# The function wraps both functions and returns the wrapped function
+
+# This examples assumes some basic familiarity with using BayBE.
+# We refer to [`campaign`](./campaign.md) for a more general and basic example.
+
+### Necessary imports for this example
+
+
+import pandas as pd
+
+from baybe import Campaign
+from baybe.objectives import SingleTargetObjective
+from baybe.objectives.base import Objective
+from baybe.parameters import NumericalDiscreteParameter
+from baybe.recommenders import RandomRecommender
+from baybe.searchspace import SearchSpace
+from baybe.targets import NumericalTarget
+from baybe.utils.basic import register_hook
+from baybe.utils.dataframe import add_fake_results
+
+### Setup
+
+# Define three test functions to test the functionality of register_hook().
+# Note that the callable function needs to have the same signature like the function from which the data are required.
+
+
+def recommend_test(
+    self,
+    batch_size: int,
+    searchspace: SearchSpace,
+    objective: Objective | None = None,
+    measurements: pd.DataFrame | None = None,
+):
+    """Print the searchspace and the batch size from the recommend call."""
+    print("start recommend_test")
+    print(searchspace, batch_size)
+    print("End recommend_test")
+
+
+### Example
+
+# We create a two phase meta recommender with default values.
+
+recommender = RandomRecommender()
+
+# We overwrite the original recommend method of the sequential greedy recommender class.
+
+RandomRecommender.recommend = register_hook(RandomRecommender.recommend, recommend_test)
+
+# We define all needen parameters for this example and collect them in a list.
+
+temperature = NumericalDiscreteParameter(
+    "Temperature", values=[90, 105, 120], tolerance=2
+)
+concentration = NumericalDiscreteParameter(
+    "Concentration", values=[0.057, 0.1, 0.153], tolerance=0.005
+)
+parameters = [temperature, concentration]
+
+# We create the searchspace and the objective.
+
+searchspace = SearchSpace.from_product(parameters=parameters)
+objective = SingleTargetObjective(target=NumericalTarget(name="yield", mode="MAX"))
+
+### Creating the campaign
+
+campaign = Campaign(
+    searchspace=searchspace,
+    recommender=recommender,
+    objective=objective,
+)
+
+# This campaign can now be used to get recommendations and add measurements:
```

Comment:
can we add anything like eg the output of the hook so that one can see that now something happened?

Reply:
You mean like adding it to the comment or adding the hook output? The hook output is actually integrated in the loop every time `.recommend` is called.

Reply:
yes but in the docupage it wont show or will it? if theres someting visible there thats fine

Reply:
Aah well the documentation pge is not made yet, I will do it after finishing the probability of improvment example, which I'm working on right now. So, I will put some output from that example not from this one. Is it oky for you?

Reply:
Currently doing my review / applying changes. Perhaps you can wait until I finished 🙃

Reply:
Fixed it. Will close this thread and give you an update what has changed in another thread.